src/api/voting.ts

```diff
@@ -184,15 +193,22 @@ export namespace VotingApi {
      * Fetch the full state of the given processId on the Vochain
      * @param processId
      * @param gateway
+     * @param params
      */
-    export function getProcessState(processId: string, gateway: IGatewayDVoteClient): Promise<ProcessState> {
+    export function getProcessState(processId: string, gateway: IGatewayClient, params: { skipArchive?: boolean } = {}): Promise<ProcessState> {
         if (!processId) return Promise.reject(new Error("Empty process ID"))
         else if (!gateway) return Promise.reject(new Error("Invalid gateway client"))
 
         return gateway.sendRequest({ method: "getProcessInfo", processId })
+            .catch((error) => {
+                const message = error.message ? "Could not retrieve the process info: " + error.message : "Could not retrieve the process info"
+                if (params.skipArchive || !message.includes("No data found for this key")) {
+                    throw new Error(message)
+                }
+                return GatewayArchiveApi.getProcess(processId, gateway, message)
```

Comment:
Don't we need to do any transformation like `mapToGetProcessState()` to adapt the archive result to the expected output?